silkworm/silkrpc/core/cached_chain.cpp

```diff
@@ -36,6 +36,32 @@ awaitable<std::shared_ptr<BlockWithHash>> read_block_by_number(BlockCache& cache
     co_return block_with_hash;
 }
 
+awaitable<std::shared_ptr<BlockWithHash>> read_block_by_number(BlockCache& cache, const std::shared_ptr<ChainStorage> storage, const rawdb::DatabaseReader& reader, uint64_t block_number) {
+    evmc::bytes32 block_hash;
+    try {
+        block_hash = co_await rawdb::read_canonical_block_hash(reader, block_number);
+    } catch (const std::invalid_argument& iv) {
+        co_return nullptr;
+    }
```

Comment:
As discussed, we should still rely on `std::invalid_argument` raised by low-level functions here without catching it at this level. Changing this behaviour consistently in the internal APIs is IMHO has broader scope than this PR.

Reply:
OK